### PR TITLE
2D obs update running version of template maker and create WS

### PR DESCRIPTION
**Pull request by ram1123**
### Summary of PR

1. This version is tested with
    1. 1D obs: `mass4l` and `pT4l`
    2. 2D obs: `massZ1 vs massZ2`
 
2. Added logger
    1. Now, instead of print we can use “[logger.info](http://logger.info/)()” or “logger.debug()”, etc
        1. Main feature is that it automatically provides filename, module name, function name and line number of the corresponding file where its called. Example:
         
        ```python
        logger.debug("Obs Name: {}\t Bins: {}".format(obsName, observableBins))
        ```
        
        Then it will give you output like below on terminal: 
        ```bash
         [DEBUG] - [runHZZFiducialXS.py:#124] - [extractBackgroundTemplatesAndFractions; runHZZFiducialXS] - [INFO] Obs Name: mass4l         	Bins: ['105.0', '140.0']
         ```
3. Added new key “ifAbs” in the YAML file to handle absolute value. Currently, its not used in the program. We can use this later at appropriate places. (Not relevant for the moment)

### To-Do:

1. Compare cards for mass4l and pT with new and old implementation
    1. Also, compare the workspaces
2. Fix the plotters... Currently, I commented them. (May be in next PR)
3. Currently, the interpolation part only run for 1D measurement. Need to discuss/think about the implementation.

### Possible next PR:

1. Remove the 8TeV WS root file calling. Its not difficult to get rid of this. It may take ~1hr to fix this.

### CODE REVIEW

**Comment by ram1123** [comment]
Update:

- I see some difference in v2 model Datacard for mass4l obs [1]. For v3 Datacards are identical.
   - From vimdiff it shows difference in the name of file.
   - Need to make sure, if its only name or also the content of file. Anyway, we need to fix both.

- for pT4l with model v2, Datacards are identical.
   -  With current update, it run for model v2 as well as v3.
   - Need to switch off v3 running for all other obs other than mass4l.

1. Screenshot 2022-03-22 at 14 59 03

**Comment by ram1123** [comment]
Hi @vukasinmilosevic ,

You can start reviewing and merge if everything looks fine.

From my side things are fine now.

About the plotter:

- LHScan and Asimov are working. I still need to update another.

I am working on it. But, we can merge this one if it looks fine for you. I will create another PR when remaining plotters are complete.

File: python/plotDifferentialBins.py [hunk 1]
```diff
@@ -238,60 +247,6 @@ def plotDifferentialBins(asimovDataModel, asimovPhysicalModel, obsName, fstate,
...
-    #     unit = ""
-    # elif (obsName=="Phi"):
-    #     label = "#Phi"
-    #     unit = ""
-    # elif (obsName=="Phi1"):
-    #     label = "#Phi_{1}"
-    #     unit = ""
-
-
     if (obsName.startswith("njets")):
```
Comment (vukasinmilosevic) [inline python/plotDifferentialBins.py:250]: Shouldn't this be also for 2D case? #24

**Review by vukasinmilosevic — APPROVED** [review]

File: python/plotDifferentialBins.py (hunk 1 above)
Comment (ram1123) [reply python/plotDifferentialBins.py:250]: yes... right.

```python
    if (obsName == "nJets" or obsName.startswith("njets") or ("jet" in obsName)):
```

This should work??

Comment (vukasinmilosevic) [reply python/plotDifferentialBins.py:250]: Will add this into another issue as you mentioned that 2D part is still not implemented here.

**Comment by vukasinmilosevic** [comment]
All seems ok with me. Locally tested with "massZ1 vs massZ2". Merging and closing the PR.

**Pull request merged**